pkg/interface/chat/src/js/components/chat.js

```diff
@@ -170,104 +145,75 @@ export class ChatScreen extends Component {
     }
   }
 
-  updateNumPeople() {
-    let conf = this.props.configs[this.state.station] || {};
-    let sis = _.get(conf, 'con.sis');
-    let numPeople = !!sis ? sis.length : 0;
-    if (numPeople !== this.state.numPeople) {
-      this.setState({ numPeople });
-    }
-  }
-
   render() {
     const { props, state } = this;
 
-    let config = props.configs[state.station] || {};
-    let messages = props.messages.slice(0);
+    let messages = props.envelopes.slice(0);
     
-    // Pending messages get pinned to the bottom of the messages queue.
-    
-    let pendingInRoom = 
-    (this.pendingQueue.has(this.state.station)) 
-    ? this.pendingQueue.get(this.state.station) : [];
-
-    pendingInRoom.map(function(value) {
-      return value.pending = true;
-    })
-
-    messages = messages.concat(pendingInRoom);
-
     let lastMsgNum = (messages.length > 0) ?
-      messages[messages.length - 1].num : 0;
+      messages.length : 0;
 
-    if (messages.length > 50 * state.numPages) {
+    if (messages.length > 100 * state.numPages) {
       messages = messages
-        .slice(messages.length - (50 * state.numPages), messages.length);
+        .slice(messages.length - (100 * state.numPages), messages.length);
     }
 
-    let reversedMessages = messages.reverse();
-    let chatMessages = reversedMessages.map((msg, i) => {
-      // Render sigil if previous message is not by the same sender
-      let gamAut = ['gam', 'aut'];    
+    let pendingMessages = 
+	    props.pendingMessages.has(state.station) 
+      ? props.pendingMessages.get(state.station) : [];
+	
+    pendingMessages.map(function(value) {
+      return value.pending = true;
+    })
```

Comment:
Does this do what you expect?  It looks like you want to set the `pending` attribute, but a comparable example doesn't seem to work: 

```
> let foo = [{ p: false, q: true }, { p: true, q: false }]
> foo.map(function(value) { return value.p = true })
[ true, true ]
```

Reply:
My bad, this *mutates* `pendingMessages` and doesn't actually care about the returned value `[true, true]`.  Gotcha.